src/Tasks/RoslynCodeTaskFactory/RoslynCodeTaskFactory.cs

```diff
@@ -573,6 +590,12 @@ internal static bool TryResolveAssemblyReferences(TaskLoggingHelper log, RoslynC
             items = hasInvalidReference ? null : resolvedAssemblyReferences.Select(i => (ITaskItem)new TaskItem(i)).ToArray();
 
             return !hasInvalidReference;
+
+            static Assembly TryLoadAssembly(string directory, string name)
```

Comment:
Is it possible to have a single method that looks at the list? That feels more robust to me.

Reply:
I thought about that, and I think the answer is "not cleanly." As I currently understand this, the fact that RoslynCodeTaskFactory is your TaskFactory is essentially a detail as part of execute a particular task. It notices that you said you wanted that, and some RoslynCodeTaskFactory code runs to set up the task for execution, but when it's finished, the task hasn't yet executed, so we can't remove the ResolveEventHandler yet. Then it executes the task in code not associated with RoslynCodeTaskFactory and comes back at the end to call a generic "cleanup" that calls this. If I want it all to be in one method as would be ideal, I think I'd have to first make all TaskFactories return a list of ResolveEventHandlers at some point (though most of them would return null), then check if it's non-null just before actually executing the task and add them to the AssemblyResolve if so and remove them afterwards. CleanupTask should always run even if the task's execution throws an exception (it's in a finally block) and briefly returns control back to the RoslynCodeTaskFactory with the relevant information, so it seemed like the best option. I did verify in a debugger that it got to that point and found a list with the expected number of ResolveEventHandlers in it and removed them all without exceptions. (I don't remember if I verified they were attached beforehand—either I did or I didn't see how to.)

Reply:
NTS: Condense various handlers into one handler